src/App.php

```diff
@@ -212,79 +204,59 @@ public static function delete(string $url): Route
      *
      * Set a callback function that will be initialized on application start
      *
-     * @param callable $callback
-     * @param array $resources
-     * @param string $group Pass "*" for all
-     *
-     * @return void
+     * @return Hook
      */
-    public static function init(callable $callback, array $resources = [], string $group = '*'): void
+    public static function init(): Hook
     {
-        if (!isset(self::$init[$group])) {
-            self::$init[$group] = [];
-        }
-
-        self::$init[$group][] = ['callback' => $callback, 'resources' => $resources];
+        $hook = new Hook();
+        $hook->groups(['*']);
+        self::$init[] = $hook;
+        return $hook;
     }
 
     /**
      * Shutdown
      *
      * Set a callback function that will be initialized on application end
      *
-     * @param callable $callback
-     * @param array $resources
-     * @param string $group Use "*" for all
-     *
-     * @return void
+     * @return Hook
      */
-    public static function shutdown(callable $callback, array $resources = [], string $group = '*'): void
+    public static function shutdown(): Hook
     {
-        if (!isset(self::$shutdown[$group])) {
-            self::$shutdown[$group] = [];
-        }
-
-        self::$shutdown[$group][] = ['callback' => $callback, 'resources' => $resources];
+        $hook = new Hook();
+        $hook->groups(['*']);
```

Comment:
Not sure if this would work all the time, any second usage of the `groups` method will overwrite the *, do we want to provide a way to not use * or always force it? not sure what we do today.

Reply:
It is the same today, default is '*' but providing the value overwrites '*', so I added this behavior, so user don't have to provide `*` when they want only this group assigned.